src/api.v2/helpers/pipeline/handleQCResponse.js

```diff
@@ -0,0 +1,147 @@
+const AWSXRay = require('aws-xray-sdk');
+const validateRequest = require('../../../utils/schema-validator');
+const AWS = require('../../../utils/requireAWS');
+const getLogger = require('../../../utils/getLogger');
+
+
+const HookRunner = require('./hooks/HookRunner');
+const assignPodToPipeline = require('./hooks/assignPodToPipeline');
+const { cleanupPods } = require('./hooks/podCleanup');
+const sendNotification = require('./hooks/sendNotification');
+
+const constants = require('./constants');
+const getPipelineStatus = require('./getPipelineStatus');
+
+const Experiment = require('../../model/Experiment');
+const Plot = require('../../model/Plot');
+
+const logger = getLogger();
+
+const hookRunner = new HookRunner();
+
+hookRunner.register(constants.ASSIGN_POD_TO_PIPELINE, [assignPodToPipeline]);
+hookRunner.registerAll([sendNotification]);
+hookRunner.register('configureEmbedding', [cleanupPods]);
+
+const getS3Output = async (message) => {
+  const { output: { bucket, key } } = message;
+  const s3 = new AWS.S3();
+  const outputObject = await s3.getObject(
+    {
+      Bucket: bucket,
+      Key: key,
+    },
+  ).promise();
+
+  return JSON.parse(outputObject.Body.toString());
+};
+
+const updatePlotData = async (taskName, experimentId, output) => {
+  if (output.plotDataKeys) {
+    const plotConfigUploads = Object.entries(output.plotDataKeys)
+      .map(async ([plotUuid, s3DataKey]) => (
+        await new Plot().upsert(
+          { id: plotUuid, experiment_id: experimentId },
+          { s3_data_key: s3DataKey },
+        )
+      ));
+
+    logger.log('Uploading plotData for', taskName, 'to DynamoDB');
+
+    // Promise.all stops if it encounters errors.
+    // This handles errors so that error in one upload does not stop others
+    await Promise.all(
+      plotConfigUploads.map((p) => p.catch((e) => logger.error(e))),
+    );
+  }
+};
+
+const updateProcessingConfig = async (taskName, experimentId, output, sampleUuid) => {
+  // TODO the processing config validation was not being enforced because the 'anyOf' requirement
+  // was not being correctly applied. This needs to be refactored together with the
+  // pipeline and ideally while unifying the qc & gem2s responses.
+  // await validateRequest(output, 'ProcessingConfigBodies.v1.yaml');
```

Comment:
do we have a ticket for this work?

Reply:
I am not sure, this comment existed from before my changes